### PR TITLE
Feature/feature like post

### PR DESCRIPTION
**Pull request by AlbertPG**

### CODE REVIEW

File: app/src/test/java/org/pesmypetcare/mypetcare/controllers/community/TestTrLikePost.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+public class TestTrLikePost {
+    private TrLikePost trLikePost;
+    private User user;
+    private Post post;
+
+    @Before
+    public void setUp() {
+        trLikePost = new TrLikePost(new StubCommunityService());
+        user = new User("Manolo Lama", "bicholover@gmail.com", "1234");
+        Group group = new Group("Husky", "John Doe",
```
Comment (santidrj) [inline app/src/test/java/org/pesmypetcare/mypetcare/controllers/community/TestTrLikePost.java:29]: Codacy Issue found: [The String "John Doe" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5443034)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Coverage increased per file
===========================
+ app/src/main/java/org/pesmypetcare/mypetcare/features/community/posts/Post.java  16
         

Complexity increasing per file
==============================
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/community/TestTrLikePost.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/community/TestTrUnlikePost.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrUnlikePost.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/services/StubCommunityService.java  12
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/circularentry/posts/PostComponentView.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/community/TrLikePost.java  1
         

Clones added
============
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/circularentry/calendar/EventComponentView.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/circularentry/petsinfo/PetsInfoComponentView.java  1
         

Clones removed
==============
+ app/src/main/java/org/pesmypetcare/mypetcare/activities/views/circularentry/calendar/EventView.java  -1
+ app/src/main/java/org/pesmypetcare/mypetcare/activities/views/circularentry/petsinfo/PetsInfoView.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5443034&bid=17848548)

File: app/src/main/java/org/pesmypetcare/mypetcare/services/StubCommunityService.java [hunk 2]
```diff
@@ -35,13 +36,13 @@
...
         addTags();
-        addFroums();
+        addForums();
     }
 
     /**
      * Add the forums.
      */
-    private static void addFroums() {
+    private static void addForums() {
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/services/StubCommunityService.java:45]: Codacy Issue found: [Method length is 17 lines (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5443034)

File: app/src/test/java/org/pesmypetcare/mypetcare/controllers/community/TestTrUnlikePost.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+ */
+public class TestTrUnlikePost {
+    private User user;
+    private Post post;
+    private TrUnlikePost trUnlikePost;
+
+    @Before
+    public void setUp() {
+        user = new User("John Smith", "johndoe@gmail.com", "1234");
+        Group group = new Group("Husky", "John Doe", DateTime.Builder.buildDateString("2020-04-15"));
```
Comment (santidrj) [inline app/src/test/java/org/pesmypetcare/mypetcare/controllers/community/TestTrUnlikePost.java:28]: Codacy Issue found: [The String "John Doe" appears 3 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5443034)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/views/circularentry/calendar/EventView.java [hunk 4]
```diff
@@ -1,71 +1,59 @@
...
-    @Override
-    protected String getFirstLineText() {
-        return pet.getName() + " - " + event.getDescription();
+    public void showEvents(Pet pet, String date) throws ParseException {
+        //pet.addEvent(new Event("Take to vet", "2020-04-03T10:30:00"));
+        List<Event> events = pet.getEvents(date);
+        List<Event> periodicEvents = pet.getPeriodicEvents(date);
+        events.addAll(periodicEvents);
+        for (Event event : events) {
+            CircularEntryView circularEntryView = new EventComponentView(context, null, pet, event)
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/views/circularentry/calendar/EventView.java:45]: Codacy Issue found: [Avoid instantiating new objects inside loops](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5443034)

**Review by enrichp — DISMISSED** [review]

**Review by xavier-campos-diaz — DISMISSED** [review]

**Review by xavier-campos-diaz — APPROVED** [review]

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ app/src/main/java/org/pesmypetcare/mypetcare/features/community/posts/Post.java  16
         

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/services/StubCommunityService.java  12
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/community/TrUnlikePost.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/circularentry/posts/PostComponentView.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/community/TestTrLikePost.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/community/TrLikePost.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/community/TestTrUnlikePost.java  1
         

Clones added
============
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/circularentry/calendar/EventComponentView.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/circularentry/petsinfo/PetsInfoComponentView.java  1
         

Clones removed
==============
+ app/src/main/java/org/pesmypetcare/mypetcare/activities/views/circularentry/calendar/EventView.java  -1
+ app/src/main/java/org/pesmypetcare/mypetcare/activities/views/circularentry/petsinfo/PetsInfoView.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5443034&bid=17848548)

**Review by enrichp — APPROVED** [review]

**Pull request merged**